### ISSUE TITLE
Keyboard interaction for Tree View 

### ISSUE THREAD

**Issue by gnomeontherun** [vmware/clarity#3569]
## Describe the bug

The Tree View does not support the correct keyboard interactions. Currently , tree nodes can be focused with the Tab key and activated with the Enter key. But the main behaviors expected for trees are as follows.

1) The entire tree structure should be focusable as a whole when using the Tab key
2) When the tree has focus, the user should be able to use the arrow keys to navigate the nodes in the tree in any direction depending on the complexity of the tree.
3) See the rest of the expected keyboard interactions at https://www.w3.org/TR/wai-aria-practices-1.1/#TreeView below.

## How to reproduce

https://stackblitz.com/github/clane/clarityTest2/tree/master/clarity-test-app

- Go to the Tree View - Basic and try to use the arrow keys to navigate the tree nodes and find that you cannot.

## Expected behavior

See the Accessible Tree example at http://whatsock.com/bootstrap/

## Versions

**App**

* Angular: [e.g. 6]
* Node: [e.g. 8.10.0]
* Clarity: [e.g. 0.12.5]

**Device:**

* Type: [e.g. MacBook]
* OS: [e.g. iOS]
* Browser [e.g. chrome, safari]
* Version [e.g. 22]

## Additional notes

_Add any other notes about the problem here._

**Comment by hippee-lee** [comment]
Possible approach is to add keypress listeners to `tree-node.html`

```
(keydown.arrowLeft)="" 
(keydown.arrowRight)="" 
(keydown.arrowDown)="" 
(keydown.arrowUp)="" 
```

Then there can be handler functions that implement the logic for deciding which node to navigate to (by finding a parent or sibling or expanding children)

**Comment by Jinnie** [comment]
Please see my comment here
https://github.com/clr-team/planning/issues/21#issuecomment-510446423
about possibility to use the KeyFocus directive as solution here.
